index.js

```diff
@@ -0,0 +1,160 @@
+      this.$emit('cardsubtitle-hidden', card);
+    },
+    requestShowSubtitle(card) {
+      this.$emit('cardsubtitle-shown', card);
+    },
+    requestAddToFavorite(card) {
+      this.$emit('added-to-favorite', card);
+    },
+    requestDeleteFromFavorite(card) {
+      this.$emit('deleted-from-favorite', card);
+    }
+  },
+}
+
+Vue.component('cards', {
+  props: {
+    cards: Array
+  },
+  components: {
+    'card': cardComponent,
+  },
+  methods: {
+    requestRemoveCard(card) {
+      this.$emit('card-removed', card);
+    },
+    requestHideSubtitle(card) {
+      this.$emit('cardsubtitle-hidden', card);
+    },
+    requestShowSubtitle(card) {
+      this.$emit('cardsubtitle-shown', card);
+    },
+    requestAddToFavorite(card) {
+      this.$emit('added-to-favorite', card);
+    },
+    requestDeleteFromFavorite(card) {
+      this.$emit('deleted-from-favorite', card);
+    }
+  },
+  template: `
+    <div class="row">
+      <card v-for="(card, index) in cards" 
+        :key="card.id" 
+        :card="card" 
+        @card-removed="requestRemoveCard(card)" 
+        @added-to-favorite="requestAddToFavorite(card)" 
+        @deleted-from-favorite="requestDeleteFromFavorite" 
+        @cardsubtitle-shown="requestShowSubtitle(card)" 
+        @cardsubtitle-hidden="requestHideSubtitle(card)">
+      </card>
+    </div>
+  `,
+  data() {
+    return {
+    };
+  }
+})
+
+new Vue({
+  el: '#app',
+  data: {
+    cards: [
+      {
+        id: 1,
+        title: 'Seitan polaroid flannel',
+        subtitle: 'next level',
+        isShowSubtitle: false,
+        isFavorite: false,
+      },
+      { id: 2, 
+        title: 'Street art swag',
+        isFavorite: false,
+      },
+      {
+        id: 3,
+        title: 'Deep v selvage tousled',
+        subtitle: 'tousled copper mug, gochujang crucifix try-hard tbh',
+        isShowSubtitle: false,
+        isFavorite: false,
+      },
+    ],
+    title: '',
+    subtitle: '',
+    count: 3,
+    inFavorites: [],
+  },
+  
+  methods: {
+    removeFirst() {
+      this.cards.shift();
+    },
+    removeLast() {
+      this.cards.pop();
+    },
+    removeCard(card) {
+      this.cards.splice(this.cards.indexOf(card), 1);
+      this.deleteFromFavorite(card);
+    },
+    hideSubtitle(card) {
+      card.isShowSubtitle = false;
+    },
+    showSubtitle(card) {
+      card.isShowSubtitle = true;
+    },
+    addCard() {
+      this.count++;      
+      this.cards.push({
+        id: this.count,
+        title: this.title,
+        subtitle: this.subtitle,
+        isShowSubtitle: false,
+        isFavorite: false
+      });
+      this.title = '';
+      this.subtitle = '';
+    },
+    addToFavorite(card) {
+      if (this.inFavorites.includes(card.id) === true) {
+        return;
+      }
+      this.inFavorites.push(card.id);
+      card.isFavorite = true;
+    },
+    deleteFromFavorite(card) {
+      if (this.inFavorites.includes(card.id) === false) {
```

Comment:
`if (!this.inFavorites.includes(card.id)) {`

Reply:
I know. Thanks. It's easier for me to read the code.